test/hotspot/jtreg/compiler/lib/compile_framework/CompileFramework.java

```diff
@@ -0,0 +1,254 @@
+        compileJavaFiles(javaFileNames);
+
+        System.out.println("Java sources compiled.");
+    }
+
+    private static void compileJavaFiles(List<String> javaFileNames) {
+        // Compile JAVA files with javac, in the "compile.jdk".
+        List<String> command = new ArrayList<>();
+
+        command.add("%s/bin/javac".formatted(System.getProperty("compile.jdk")));
+        command.add("-classpath");
+        command.add(System.getProperty("java.class.path"));
+        command.add("-d");
+        command.add(System.getProperty("test.classes"));
+        for (String fileName : javaFileNames) {
+            command.add(new File(fileName).getAbsolutePath());
+        }
+
+        executeCompileCommand(command);
+    }
+
+    private static List<String> writeSourcesToFile(String sourceDir, List<SourceCode> sources) {
+        List<String> storedFiles = new ArrayList<String>();
+        for (SourceCode sourceCode : sources) {
+            String extension = sourceCode.kind.name().toLowerCase();
+            String fileName = sourceDir + "/" + sourceCode.className.replace('.','/') + "." + extension;
+            writeCodeToFile(sourceCode.code, fileName);
+            storedFiles.add(fileName);
+        }
+        return storedFiles;
+    }
+
+    private static void writeCodeToFile(String code, String fileName) {
+        System.out.println("File: " + fileName);
+        File file = new File(fileName);
+        File dir = file.getAbsoluteFile().getParentFile();
+        if (!dir.exists()){
+            dir.mkdirs();
+        }
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(file))) {
+            writer.write(code);
+        } catch (Exception e) {
+            throw new CompileFrameworkException("Could not write file: " + fileName, e);
+        }
+    }
+
+    private static void executeCompileCommand(List<String> command) {
+        System.out.println("Compile command: " + String.join(" ", command));
+
+        ProcessBuilder builder = new ProcessBuilder(command);
+        builder.redirectErrorStream(true);
+
+        String output;
+        int exitCode;
+        try {
+            Process process = builder.start();
+            boolean exited = process.waitFor(JASM_COMPILE_TIMEOUT, TimeUnit.SECONDS);
+            if (!exited) {
+                process.destroyForcibly();
+                throw new InternalCompileFrameworkException("Process timeout: compilation took too long.");
+            }
+            output = new String(process.getInputStream().readAllBytes(), StandardCharsets.UTF_8);
+            exitCode = process.exitValue();
+        } catch (IOException e) {
+            throw new InternalCompileFrameworkException("IOException during compilation", e);
+        } catch (InterruptedException e) {
+            throw new CompileFrameworkException("InterruptedException during compilation", e);
+        }
+
+        if (exitCode != 0 || !output.equals("")) {
```

Comment:
> Note: FuzzerUtils.java uses or overrides a deprecated API.
> Note: Recompile with -Xlint:deprecation for details.

Warnings could corrupt the output. And we probably need to at least make it controllable.

Reply:
I see. I have so far not encountered any issues. I would like to keep it simple for now. We can invest in a more complicated solution in a follow up RFE. For now I suppose `FuzzerUtils` would just not be allowed.

Reply:
You can probably replace this by:
```suggestion
        if (exitCode != 0 || !output.isEmpty()) {
```